src/devices/bus/ss50/mps.cpp

```diff
@@ -48,26 +50,34 @@ class ss50_mps_device : public device_t, public ss50_card_interface
 
 private:
 	DECLARE_WRITE_LINE_MEMBER(acia_irq_w);
+	DECLARE_WRITE_LINE_MEMBER(route_cts);
 
 	required_device<acia6850_device> m_acia;
 	required_ioport m_irq_jumper;
 	required_ioport m_rate_jumper;
+	required_ioport m_cts_route;
 };
 
 
 static INPUT_PORTS_START( mps )
 	PORT_START("IRQ")
-	PORT_DIPNAME(1, 0, "IRQ")
-	PORT_DIPSETTING(1, DEF_STR(Off))
-	PORT_DIPSETTING(0, DEF_STR(On))
+	PORT_CONFNAME(1, 0, "IRQ")
+	PORT_CONFSETTING(1, DEF_STR(Off))
+	PORT_CONFSETTING(0, DEF_STR(On))
 
 	PORT_START("BAUD")
-	PORT_DIPNAME(0x1f, 0x1d, "Baud Rate")
-	PORT_DIPSETTING(0x1e, "110")
-	PORT_DIPSETTING(0x1d, "150 / 9600")
-	PORT_DIPSETTING(0x1b, "300")
-	PORT_DIPSETTING(0x17, "600")
-	PORT_DIPSETTING(0x0f, "1200")
+	PORT_CONFNAME(0x1f, 0x1d, "Baud Rate")
+	PORT_CONFSETTING(0x1e, "110")
+	PORT_CONFSETTING(0x1d, "150 / 9600")
+	PORT_CONFSETTING(0x1b, "300")
+	PORT_CONFSETTING(0x17, "600")
+	PORT_CONFSETTING(0x0f, "1200")
+
+	PORT_START("CTS_ROUTE")
+	PORT_CONFNAME(1, 0, "CTS route")
+	PORT_CONFSETTING(0, "Wired low (standard)")
+	PORT_CONFSETTING(1, "Connected through (modified)")
```

Comment:
This should have a `PORT_CHANGED_MEMBER` as well for doing live changes.

Reply:
These don't a `PORT_CHANGED_MEMBER` as the port is re-read each time a decision is made. I had been pondering if that was too heavy, and if a separate variable is needed, and then the `PORT_CHANGED_MEMBER` would be needed to update that state. For the control lines, their state is not expected to be changing at a high frequency, so this seems ok. What do you think?

Reply:
You need a port changed member because you're not going to immediately get an update from the MCU when the switch is changed.

When the switch is switched to route, you need to update the output from the fixed value to whatever the terminal last output.  Otherwise it won't update until the next time the terminal changes its output.

When the switch is changed to don't route, you need to update the output from whatever the terminal last output to the fixed value.  Otherwise you'll never update it because you ignore the terminal's output when the switch is set not to route.

Reply:
Great point, thanks. Let me rework that.